homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,137 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    entities = [ZimiLight(device, api) for device in api.lights]
+
+    async_add_entities(entities)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        self._state = False
+        self._brightness = None
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._light.name,
+            suggested_area=self._light.room,
+            via_device=(DOMAIN, api.mac),
+        )
+        self.update()
+        _LOGGER.debug("Initialising %s in %s", self.name, self._light.room)
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        return self._light.name.strip()
+
+    @property
+    def available(self) -> bool:
+        """Return True if Home Assistant is able to read the state and control the underlying device."""
+        return self._light.is_connected
```

Comment:
better to set `_attr_name` in init. Apologies, I think we already had this discussion somewhere, is this not the main feature of the device?

Reply:
We did discuss this but I thought it resolved as I didn't understand I should use _attr_name.   Now changed and name property removed.   Tested and works.

https://github.com/home-assistant/core/pull/129876/commits/bc514d08fbad57233a3bed4b2a91c82d26f8fbae